test/e2e/functional/tests/e2e.go

```diff
@@ -3,355 +3,895 @@ package tests
-				Expect(err).ToNot(HaveOccurred())
-				Expect(len(pods.Items)).To(Equal(int(daemonset.Status.DesiredNumberScheduled)))
-
-				for _, pod := range pods.Items {
-					Expect(pod.Status.Phase).To(Equal(corev1.PodRunning))
-				}
-			})
-			By("checking Ingress Node Firewall Config CR status is set", func() {
-				Eventually(func() bool {
-					err := testclient.Client.Get(context.Background(), goclient.ObjectKey{Namespace: config.Namespace, Name: config.Name}, config)
-					Expect(err).ToNot(HaveOccurred())
-					if config.Status.Conditions == nil {
-						return false
-					}
-					for _, condition := range config.Status.Conditions {
-						switch condition.Type {
-						case status.ConditionAvailable:
-							if condition.Status == metav1.ConditionFalse {
-								return false
-							}
-						case status.ConditionProgressing:
-							if condition.Status == metav1.ConditionTrue {
-								return false
-							}
-						case status.ConditionDegraded:
-							if condition.Status == metav1.ConditionTrue {
-								return false
-							}
-						}
-					}
-					return true
-				}, 5*time.Minute, 5*time.Second).Should(BeTrue())
-			})
-		})
-
-		It("should run Ingress node firewall apply rules and check the actions", func() {
-			By("get nodes IP addresses with matching labels and ping their IPs")
-			var rules *ingressnodefwv1alpha1.IngressNodeFirewall
-			nodes, err := testclient.Client.Nodes().List(context.Background(), metav1.ListOptions{LabelSelector: consts.IngressNodeFirewallNodeLabel})
-			Expect(err).ToNot(HaveOccurred())
-
-			err, _ = infwutils.RunPingTest(nodes.Items)
-			Expect(err).ToNot(HaveOccurred())
-
-			v4CIDR, v6CIDR, err := infwutils.GetRuleCIDR(TestIsSingleStack, nodes.Items)
-			Expect(err).ToNot(HaveOccurred())
-
-			By("creating ingress node firewall rules")
-			if !TestIsSingleStack && v4CIDR != "" && v6CIDR != "" {
-				rules = &ingressnodefwv1alpha1.IngressNodeFirewall{
-					ObjectMeta: metav1.ObjectMeta{
-						Name: "rules1",
+		// We test IngressNodeFirewall configuration against pod to pod communication in a cluster, that is, our source CIDRs
+		// must be a subset of the pod network CIDR.
+		// The reason we are testing pod to pod communication, which isn't a typical use case for INF,
+		// is to minimise possibility of cluster disruption during test runs, and it also gives us flexibility to create
+		// (and disrupt) as many connections as we wish - only limited by the number of pods we can create.
+		// For a KinD cluster, pod to pod communication is simple and the pod IPs can be filtered on the primary interface 'eth0'.
+		// For OCP, pod IPs are visible on the GENEVE interface (with OVN-Kubernetes as the CNI) or vlan interface (with Openshift-SDN as the CNI).
+		// This is where we want to attach our XDP program and perform filtering.
+		// For OCP with OVN-Kubernetes as the CNI, the interface name is always 'genev_sys_6081'.
+		table := []struct {
```

Comment:
BTW when I originally mentioned tables I was referring to https://onsi.github.io/ginkgo/#table-specs
since it was my mistake for lack of clarity I can convert your tables to ginko once ur changes are merged